### PR TITLE
NTV-170 :Segment Analytics Call

### PR DESCRIPTION
**Pull request by hadia**
# 📲 What

Add a new event to the confirmation CTA,New event has context_cta=pledge_confirm

# 🛠 How
Screen Shot 2021-09-17 at 9 54 09 PM
# 👀 See
Screen Shot 2021-09-17 at 9 21 55 PM

# 📋 QA

Pledge project and click confirm Button

# Story 📖
https://kickstarter.atlassian.net/browse/NTV-170

### CODE REVIEW

File: app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt [hunk 1]
```diff
@@ -1391,6 +1382,18 @@ interface PledgeFragmentViewModel {
...
 
+            this.onRiskManagementConfirmed
+                .compose<Pair<Void, PledgeReason>>(combineLatestPair(pledgeReason))
+                .filter {
+                    it.second == PledgeReason.PLEDGE
+                }
+                .compose(combineLatestPair(checkoutAndPledgeData))
+                .filter { it.second.second.pledgeFlowContext() == PledgeFlowContext.NEW_PLEDGE }
+                .subscribe {
+                    this.analyticEvents.trackPledgeConfirmCTA(it.second.first, it.second.second)
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt:1393]: Could you add on the PledgeFragmentViewModelTest https://github.com/kickstarter/android-oss/blob/f6d5288901a94ebdfe278e30a07acc646bc5881b/app/src/test/java/com/kickstarter/viewmodels/PledgeFragmentViewModelTest.kt#L2117

the assert for the analytics as well? 🙏

Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt:1393]: Could we add the test for the event being sent on the PledgeFragmentViewModelTest as well? 🙏  
https://github.com/kickstarter/android-oss/blob/f6d5288901a94ebdfe278e30a07acc646bc5881b/app/src/test/java/com/kickstarter/viewmodels/PledgeFragmentViewModelTest.kt#L2117

Comment (hadia) [reply app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt:1393]: @Arkariang Added ☺️

**Review by Arkariang — APPROVED** [review]
Awesome work! 💖

**Pull request merged**